source/menus/amount_menu.cpp

```diff
@@ -83,11 +85,8 @@ void AmountMenu::draw() {
         break;
     case MAGIC_INDEX:
         Cursor::moveList(g_magicNum);
-        if (g_magicNum == 0xFF) {
-            g_magicNum = 32;
-        }
-        if (g_magicNum > 32) {
-            g_magicNum = 0;
+        if (g_magicNum > g_maxmagicNum) {
+            g_magicNum = g_maxmagicNum;
         }

```

Comment:
If the player has 0 magic and they press d-pad left, the magic count will wrap back around to the maximum magic.
If the player has the maximum magic and they press d-pad right, their magic count will stay the same.

I do not think it makes sense to let the magic count wrap around when counting down but not when counting up.

Reply:
The concern I had did not get resolved.  Now, the opposite problem is happening.  Making the magic bigger than 32 makes the magic wrap back around to 0, but the magic stays stuck at 0 if the player presses d-pad left. 

Either the magic count should wrap around both ways or not at all.

Reply:
```if (g_magicNum == 0xFF) {
            g_magicNum = g_maxMagicNum;
        }
        if (g_magicNum > g_maxMagicNum) {
            g_magicNum = 0;
        }
```

Reply:
Define g_maxMagicNum with the other definitions:
```
u8 g_maxMagicNum = dComIfGs_getMaxMagic();
```